src/main/java/com/sprints/interceptors/UndeclaredParamsHandlerInterceptor.java

```diff
@@ -0,0 +1,63 @@
+package com.sprints.interceptors;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.core.DefaultParameterNameDiscoverer;
+import org.springframework.core.MethodParameter;
+import org.springframework.core.ParameterNameDiscoverer;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+import com.sprints.exception.UndeclaredRequestParamException;
+
+import org.springframework.util.StringUtils;
+
+
+
+public class UndeclaredParamsHandlerInterceptor extends HandlerInterceptorAdapter{
```

Comment:
What is the objective of this class ?

Reply:
There I implement methods like preHandle to get the parameters sent in the Get operation filters by the user and validate them before they get to the controller. Also it contains the method to get the names of valid params and compares it with the values sent by the user and throws an exception if they´re invalid